toolkit/tools/internal/pkgjson/pkgjson.go

```diff
@@ -73,11 +74,48 @@ type Package struct {
 	RunTests      bool          `json:"RunTests"`      // Should we run tests for this package.
 }
 
+func packageLess(a, b PackageVer) bool {
+	if a.Name == b.Name {
+		v1 := versioncompare.New(a.Version)
+		v2 := versioncompare.New(b.Version)
+		return v1.Compare(v2) < 0
+	}
+
+	return a.Name < b.Name
+}
+
+func SortPackageList(packages []*Package) {
+
+	sort.Slice(packages, func(i, j int) bool {
+		return packageLess(*packages[i].Provides, *packages[j].Provides)
+	})
+
+	// For each package, also sort the package lists
```

Comment:
Since we're already adding a comment, I think "dependencies lists" would be more precise and give more info to the reader. Or "its dependencies". What do you think?
```suggestion
	// For each package, also sort its dependencies
```